src/hooks/useDeployment.ts

```diff
@@ -0,0 +1,98 @@
+import { useSession } from "next-auth/react";
+import { useState, useCallback, useMemo } from "react";
+
+import { MaciService } from "~/services/maci";
+import { useEthersSigner } from "./useEthersSigner";
+import { eas } from "~/config";
+import * as vks from "../../public/vk_10-2-1-2_test.0.json";
+
+interface IDeployArgs {
+  amount: number;
+}
+
+export function useDeployment() {
+  /**
+   * deployStatus:
+   * 0. default
+   * 1. VoiceCreditProxy
+   * 2. Gatekeeper
+   * 3. Verifier
+   * 4. PollFactory
+   * 5. MessageProcessorFactory
+   * 6. TallyFactory
+   * 7. MACI contract
+   * 8. VkRegistry
+   */
+  const [deployStatus, setDeployStatus] = useState<number>(0);
+  const signer = useEthersSigner();
+  const { data } = useSession();
+
+  const maci = useMemo(
+    () => (signer ? new MaciService(signer) : undefined),
+    [Boolean(signer), useEthersSigner],
+  );
+
+  const deploy = useCallback(
+    async ({ amount }: IDeployArgs) => {
+      if (!signer || !maci) throw new Error("no signer");
+
+      const attester = await signer.getAddress();
+
+      // 1. deploy voice credit proxy contract
+      setDeployStatus(1);
```

Comment:
should we not set the status after the contract is deployed?

Reply:
actually this is mainly for me debugging, should be deployment wizard mentioned by Anton above
(which means, it should be a complete deployStatus, including "what contract is deploying now", "what transaction needs to be signed", "isLoading or successful or fail")

Reply:
ok makes sense